cmd/src/batch_exec.go

```diff
@@ -101,7 +101,7 @@ Examples:
 		}
 
 		ctx, cancel := contextCancelOnInterrupt(context.Background())
-		defer cancel()
+		defer cancel(nil)
```

Comment:
I wonder if we should be passing in nil directly here, an error could occur at any point in the execution of a batch spec.

Is there a reason why the `handler` function doesn't return a named return value and we can then pass that into `defer cancel`?

```go
handler := func(args []string) (err error) {
	...
	ctx, cancel := contextCancelOnInterrupt(context.Background())
	defer cancel(err)
	...
}
```

This way we are guaranteed to always have access to whatever error occurs in the lifecycle of an execution.